src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -92,19 +97,67 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     float professorScore = professorSentiment.getScore();
     profLanguageService.close();
 
-    Entity professorEntity = new Entity("Professor");
-    professorEntity.setProperty("comments-professor", professorFeedback);
-    professorEntity.setProperty("score-professor", professorScore);
-    professorEntity.setProperty("perception", professorRating);
+    // Gets user email.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
 
-    Entity commentsEntity = new Entity("Comments");
-    commentsEntity.setProperty("comments-class", classFeedback);
-    commentsEntity.setProperty("score-class", classScore);
-    commentsEntity.setProperty("perception", classRating);
+    // Check whether user has reviewer ID in system.
+    Filter userFilter = new FilterPredicate("user-email", FilterOperator.EQUAL, userEmail);
+    Query userQuery = new Query("User").setFilter(userFilter);
+    List<Entity> userQueryList =
+        datastore.prepare(userQuery).asList(FetchOptions.Builder.withDefaults());
+
+    // User has not reviewed any rating.
+    if (userQueryList.size() == 0) {
+      Entity newReviewer = new Entity("User");
+      newReviewer.setProperty("user-email", userEmail);
+      Entity userID = newReviewer;
+    } else {
+      Entity userID = userQueryList.get(0);
+    }
+
+    // We would query with both professor/term filer and property filter
+    // so we would search specifically whether a rating with such reviewer-id
+    // exists.
+    Query<Entity> reviewerQuery =
+        Query.newEntityQueryBuilder()
+            .setKind("Rating")
+            .setFilter(CompositeFilter.and(PropertyFilter.eq("reviewer-id", userID.getKey())))
+            .build();
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentsEntity);
-    datastore.put(professorEntity);
+    if (reviewerQuery != null) {
```

Comment:
I think you'd have to do datastore.run(reviewerQuery) before you could see the results

Reply:
I did something similar I believe, lmk if you believe what I did works!